openlcs/libs/corgi.py

```diff
@@ -32,6 +35,42 @@ def __init__(self, base_url=None):
             # corgi api endpoint available in environment variable
             base_url = os.getenv("CORGI_API_PROD")
         self.base_url = base_url
+        # corgi stage support this at the moment.
+        self.default_exclude_fields = (
+            "products",
+            "product_versions",
+            "product_streams",
+            "product_variants",
+            "channels",
+        )
+        self.session = requests.Session()
+
+    def get(self, url, query_params=None, excludes=None, timeout=30,
+            max_retries=5, retry_delay=10):
+        if not query_params:
+            query_params = {}
+        if excludes is None:
+            excludes = self.default_exclude_fields
+        # work around for latencies by excluding costly related field
+        # queries, see also CORGI-482
+        query_params['exclude_fields'] = ','.join(excludes)
+
+        for i in range(max_retries + 1):
```

Comment:
The reason I added this is that while testing this against corgi-stage, I see various 502/504 during get. It appears there's some perf regression on corgi end.

There's no guarentee a `requests.get` always works, thus retry is necessary.